go-controller/pkg/node/management-port_linux.go

```diff
@@ -280,6 +293,108 @@ func createPlatformManagementPort(interfaceName string, localSubnets []*net.IPNe
 	return cfg, nil
 }
 
+func getIPTablesForHostSubnets(hostSubnets []*net.IPNet) (util.IPTablesHelper, util.IPTablesHelper, error) {
+	var ipt4, ipt6 util.IPTablesHelper
+	var err error
+
+	for _, hostSubnet := range hostSubnets {
+		if utilnet.IsIPv6CIDR(hostSubnet) {
+			if ipt6 != nil {
+				continue
+			}
+			ipt6, err = util.GetIPTablesHelper(iptables.ProtocolIPv6)
+		} else {
+			if ipt4 != nil {
+				continue
+			}
+			ipt4, err = util.GetIPTablesHelper(iptables.ProtocolIPv4)
+		}
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+	return ipt4, ipt6, nil
+}
+
+func checkExistingPlatformMgmtPort(hostSubnets []*net.IPNet, mgmtPortName string) error {
+	_, stderr, err := util.RunOVSVsctl("list", "Interface", mgmtPortName)
+	if err != nil {
+		if !strings.Contains(stderr, "no row \""+mgmtPortName+"\" in table") {
+			return fmt.Errorf("failed to lookup OVS port: %s", stderr)
+		}
+		// Not found on the bridge. But could be that interface with the same name exists
+		return unconfigurePlatformMgmtInterface(hostSubnets, mgmtPortName)
+	}
+
+	// Found existing port. Check its type
+	stdout, stderr, err := util.RunOVSVsctl("get", "Interface", mgmtPortName, "type")
+	if err != nil {
+		return fmt.Errorf("failed to get OVS interface type: %s", stderr)
+	}
+	if stdout == "internal" {
+		klog.Infof("Found OVS internal port. Removing it")
+		_, stderr, err := util.RunOVSVsctl("del-port", "br-int", mgmtPortName)
+		if err != nil {
+			return fmt.Errorf("failed to remove OVS internal port: %s", stderr)
+		}
+		return nil
+	}
+
+	// It is other representor which was used as management port.
+	// Remove it from the bridge and rename.
+	klog.Infof("Found existing representor management port. Removing it")
+	return unconfigureMgmtRepresentorPort(mgmtPortName)
+}
+
+func unconfigureMgmtRepresentorPort(mgmtPortName string) error {
+	_, stderr, err := util.RunOVSVsctl("del-port", "br-int", mgmtPortName)
+	if err != nil {
+		return fmt.Errorf("failed to remove OVS port: %s", stderr)
+	}
+
+	link, err := util.GetNetLinkOps().LinkByName(mgmtPortName)
+	if err != nil {
+		return fmt.Errorf("failed to lookup %s link: %v", mgmtPortName, err)
+	}
+
+	if err := util.GetNetLinkOps().LinkSetDown(link); err != nil {
+		return fmt.Errorf("failed to set link down: %v", err)
+	}
+
+	// rename to PortName + "-ddmmyyHHMMSS"
+	if err := util.GetNetLinkOps().LinkSetName(link, mgmtPortName+time.Now().Format("-010206150405")); err != nil {
```

Comment:
Is the actual netdev or representor dev renamed here? or does it become orphaned device?

Reply:
Only representor can be found on the bridge. So it is renamed to avoid conflict.
Didn't get Q. about orphaned device.

Reply:
I was wondering whether the renamed devices become useless. 
If I read it correctly, both devices are renamed. The representor device is renamed in `unconfigureMgmtRepresentorPort` and netdev device is renamed in `unconfigurePlatformMgmtInterface`.  The renamed device pair can be discovered by sriov operator and reused for workload pods. do you agree?

Reply:
Can't say about operator, but sriov-device-plugin look only onto device IDs (VFs - PCI, SFs - aux.names), so netdevice will be added into the resource pool. Ovnkube, in turns, will discover representor, whichever name it has.

Reply:
The only problem, which actually can be, is when 1st pair of devices were used. Then 2nd one -> 1st pair renamed to templated name. Then user switched back to the 1st pair, but ovnkube fails to find devices under original names either mgmt port names.

Reply:
> The only problem, which actually can be, is when 1st pair of devices were used. Then 2nd one -> 1st pair renamed to templated name. Then user switched back to the 1st pair, but ovnkube fails to find devices under original names either mgmt port names.

Does it make sense to rename the device to its original name by using external-ids?

Reply:
I'd thought about it. Probably yes. Limitation here is that for DPU-host mode it is not possible. Perhaps it'll be a bit confusing to the user, that in full and DPU modes original names are restored, but not DPU-host. Let me check how it will work/look for full and DPU modes.